Class And Batch UI for Review

kartiks26 opened the pull request:
Class and batch Ui is part where user will enroll in a batch which contains 28 classes pre enrolled and every exercise / pathway course will have random,classes for same generated by respective teacher .
Upcomming batches will shown only on login and exercises are also now on will be logged in Based.
In this PR the most of work is on Pathway Course .

1. /pathway/1
![image](https://user-images.githubusercontent.com/84696578/172128725-611d1a31-f754-4831-bafc-e5ea8176746c.png)
![image](https://user-images.githubusercontent.com/84696578/172129018-593906b5-3053-4b6e-b17e-4563b6a2dd0b.png)
![image](https://user-images.githubusercontent.com/84696578/172129034-5978fc76-098c-49ef-a8b4-2dbc2035b78e.png)
![image](https://user-images.githubusercontent.com/84696578/172129286-194f728a-ae34-455e-a3c0-d9a0f8d9684c.png)


2. /course-content/1/370/1 
![image](https://user-images.githubusercontent.com/84696578/172128647-3c61f281-622c-4c64-a20a-ef4973152c8b.png)
![image](https://user-images.githubusercontent.com/84696578/172129499-42278890-bc04-44a0-9816-7a17d658cc5f.png)


3. 
![image](https://user-images.githubusercontent.com/84696578/172128616-e036a8ef-e691-4fb5-acf4-ed0f66b03b99.png)

4. /profile
![image](https://user-images.githubusercontent.com/84696578/172128837-7bf256ac-7822-4808-ae48-c30ca82570ac.png)

5. Popups 
![image](https://user-images.githubusercontent.com/84696578/172128960-88710914-140e-460f-8384-596f182ded69.png)
![image](https://user-images.githubusercontent.com/84696578/172129321-62eadce5-4b99-4f4b-8caa-c388db14fb5a.png)

6. user Dashboard
![image](https://user-images.githubusercontent.com/84696578/172129101-eb0b22ae-08b5-48a5-a9dd-10decd487b92.png)

7. Work in progress for LEarning track which is also part of this update.

Note : in Exercise Content Might Crash As work progress on Learning Track 


@jschanker

amansharmma committed:
Merge branch 'dev' of https://github.com/navgurukul/bhanwari-devi into volunteerDashbord

amansharmma committed:
Filter work is pending

amansharmma committed:
Merge branch 'dev' of https://github.com/navgurukul/bhanwari-devi into newUserDashbord

Poonam-Singh-Bagh committed:
assessment

Poonam-Singh-Bagh committed:
assessment

amansharmma committed:
Started UserDashboared

amansharmma committed:
Commit  resolved

amansharmma committed:
newUserDashbord Card issue resolved

amansharmma committed:
Merge branch 'dev' of https://github.com/navgurukul/bhanwari-devi into newUserDashbord

kartiks26 committed:
Merge pull request #324 from navgurukul/newUserDashbord

New user dashbord

amansharmma committed:
UpcomingCourse

amansharmma committed:
Merge branch 'main' into PythonCourse

kartiks26 committed:
Merge pull request #335 from navgurukul/PythonCourse

upcoming classes

Poonam-Singh-Bagh committed:
Merge branch 'main' of https://github.com/navgurukul/bhanwari-devi into assessment

kartiks26 committed:
update on aman stuff

kartiks26 committed:
Merge branch 'dev' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

kartiks26 committed:
made api integration in ui made by aman

kartiks26 committed:
Merge pull request #337 from navgurukul/uiUpdateV2

UI update v2

amansharmma committed:
UI updated

Poonam-Singh-Bagh committed:
Merge branch 'main' of https://github.com/navgurukul/bhanwari-devi into assessment

Apekshahande committed:
made cards

amansharmma committed:
Button fixed for pathway/1

kartiks26 committed:
Merge pull request #345 from navgurukul/PythonCourse

Python course

kartiks26 committed:
Update on Aman Changes

kartiks26 committed:
Merge branch 'main' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

kartiks26 committed:
Merge pull request #346 from navgurukul/uiUpdateV2

UI update v2

Apekshahande committed:
Merge branch 'dev' of https://github.com/navgurukul/bhanwari-devi into pathwaycards

kartiks26 committed:
Merge pull request #341 from navgurukul/pathwaycards

PathwayCards

amansharmma committed:
Ui fixed

amansharmma committed:
Ui fixed

amansharmma committed:
Ui fixed

amansharmma committed:
Merge branch 'dev' of https://github.com/navgurukul/bhanwari-devi into PythonCourse

amansharmma committed:
created join clsss card

kartiks26 committed:
Merge pull request #349 from navgurukul/PythonCourse

Python course

amansharmma committed:
PathwayExercise

amansharmma committed:
Merge branch 'dev' of https://github.com/navgurukul/bhanwari-devi into PythonCourse

amansharmma committed:
EnrollCourse component completed

amansharmma committed:
Revison class component completed

amansharmma committed:
added help box in the ui

kartiks26 committed:
Merge pull request #363 from navgurukul/PythonCourse

Python course

kartiks26 committed:
added api and stuff

kartiks26 committed:
Merge branch 'dev' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

Poonam-Singh-Bagh committed:
assessment

kartiks26 committed:
fixed aman ui in main ui with correct positions.

kartiks26 committed:
Merge pull request #365 from navgurukul/uiUpdateV2

UI update v2

kartiks26 committed:
update on class ui

kartiks26 committed:
Merge pull request #379 from navgurukul/uiUpdateV2

update on class ui

kartiks26 committed:
Merge branch 'main' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

kartiks26 committed:
updated auth

kartiks26 committed:
Merge pull request #380 from navgurukul/uiUpdateV2

UI update v2

amansharmma committed:
SideBar Created into the revisionClassExercise

amansharmma committed:
SideBar Created into the revisionClassExercise

kartiks26 committed:
Merge pull request #382 from navgurukul/BatchOfMoreDetailesSideBar

Batch of more detailes side bar

kartiks26 committed:
not enrolled student cant see stuff

kartiks26 committed:
Merge branch 'dev' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

Poonam-Singh-Bagh committed:
assessment

kartiks26 committed:
updated not logged in part, changed date format

kartiks26 committed:
Merge pull request #387 from navgurukul/uiUpdateV2

UI update v2

Poonam-Singh-Bagh committed:
assessment

Poonam-Singh-Bagh committed:
assessment conflicts resolved

Poonam-Singh-Bagh committed:
assessment

kartiks26 committed:
Added time left disabled , class ui batch componen

Poonam-Singh-Bagh committed:
assessment

kartiks26 committed:
Merge pull request #389 from navgurukul/uiUpdateV2

Added time left disabled , class ui batch componen

Poonam-Singh-Bagh committed:
enable delete class feature

Poonam-Singh-Bagh committed:
assessment

Poonam-Singh-Bagh committed:
assessment re-try

kartiks26 committed:
updated revision class id

kartiks26 committed:
added Ui features

kartiks26 committed:
Merge branch 'main' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

kartiks26 committed:
Merge branch 'main' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

kartiks26 committed:
Merge pull request #392 from navgurukul/uiUpdateV2

UI update v2

kartiks26 committed:
made route private looked over making the mobile v.

Poonam-Singh-Bagh committed:
assessment

Poonam-Singh-Bagh committed:
assessment

kartiks26 committed:
completed batch UI

kartiks26 committed:
Merge pull request #395 from navgurukul/uiUpdateV2

UI update v2

amansharmma committed:
work Done for upcomming classes card

kartiks26 committed:
Merge pull request #396 from navgurukul/ClassesCard

Classes card

kartiks26 committed:
updated file structure

kartiks26 committed:
added Profile dropOut , added other flows and code

kartiks26 committed:
Merge pull request #397 from navgurukul/uiUpdateV2

UI update v2

kartiks26 committed:
Updated Fixes

kartiks26 committed:
Merge pull request #398 from navgurukul/uiUpdateV2

UI update v2

Commit:
add Batch change time

Commit:
css add

Commit:
range button droup

amansharmma committed:
work done for returning user

kartiks26 committed:
Merge pull request #400 from navgurukul/Banner

Banner

kartiks26 committed:
Merge pull request #402 from navgurukul/updateCheckbox

Update checkbox

kartiks26 committed:
fixed Some bugs
Merged Aman saloni changes

kartiks26 committed:
Merge pull request #403 from navgurukul/uiUpdateV2

fixed Some bugs

amansharmma committed:
added

amansharmma committed:
done

amansharmma committed:
added heading

amansharmma committed:
Work done for Upcoming Doubt class section

kartiks26 committed:
Merge pull request #405 from navgurukul/DoubtClass

Work done for Upcoming Doubt class section

amansharmma committed:
added heading

kartiks26 committed:
Merge pull request #407 from navgurukul/Banner

Banner

kartiks26 committed:
fixed some issues and improvement in code

kartiks26 committed:
Merge pull request #408 from navgurukul/uiUpdateV2

fixed some issues and improvement in code

amansharmma committed:
Work done for feedback

amansharmma committed:
Work done for feedback of doubt class section

kartiks26 committed:
Merge pull request #409 from navgurukul/DoubtClass

Doubt class

kartiks26 committed:
fiexd some bugs

kartiks26 committed:
Merge pull request #410 from navgurukul/uiUpdateV2

fiexd some bugs

amansharmma committed:
Adding Course

kartiks26 committed:
merged main to dev

Poonam-Singh-Bagh committed:
Merge branch 'assessment' of https://github.com/navgurukul/bhanwari-devi into NewFeature

kartiks26 committed:
Merge branch 'assessment' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

kartiks26 committed:
Merge pull request #411 from navgurukul/uiUpdateV2

UI update v2

kartiks26 committed:
fixed Minor Bug

kartiks26 committed:
Merge pull request #412 from navgurukul/uiUpdateV2

fixed Minor Bug

Poonam-Singh-Bagh committed:
merged Assessment and classes and batches

kartiks26 committed:
Merge branch 'Banner' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

kartiks26 committed:
fixed Returning User Thing

kartiks26 committed:
Merge pull request #413 from navgurukul/uiUpdateV2

UI update v2

kartiks26 committed:
Merge branch 'NewFeature' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

kartiks26 committed:
Merge pull request #414 from navgurukul/uiUpdateV2

UI update v2

kartiks26 committed:
added dateTimeFormat and TimeLeft in Try catch

kartiks26 committed:
Merge pull request #415 from navgurukul/uiUpdateV2

added dateTimeFormat and TimeLeft in Try catch

Commit:
open-course

Poonam-Singh-Bagh committed:
added /enrolledBatches api to the redux

Poonam-Singh-Bagh committed:
assessment

Poonam-Singh-Bagh committed:
assessment

Poonam-Singh-Bagh committed:
Merge pull request #417 from navgurukul/NewFeature

New feature

Commit:
NoBatchEnroll

kartiks26 committed:
Merge branch 'batchUiFixes' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

kartiks26 committed:
fixes

Poonam-Singh-Bagh committed:
add /enrolledBatches in redux

kartiks26 committed:
Merge branch 'dev' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

kartiks26 committed:
ui fixes

kartiks26 committed:
Merge pull request #418 from navgurukul/uiUpdateV2

UI update v2

kartiks26 committed:
finished working on batch UI, fixed some flow

kartiks26 committed:
Merge pull request #420 from navgurukul/uiUpdateV2

finished working on batch UI, fixed some flow

amansharmma committed:
fixed some errors

kartiks26 committed:
Merge pull request #421 from navgurukul/SideBarUpdate

Side bar update

Poonam-Singh-Bagh committed:
deployed on vercel

Poonam-Singh-Bagh committed:
Merge branch 'dev' of https://github.com/navgurukul/bhanwari-devi into dev

kartiks26 committed:
improvments and bug fixes

kartiks26 committed:
Merge branch 'dev' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

kartiks26 committed:
Merge pull request #424 from navgurukul/uiUpdateV2

improvments and bug fixes

kartiks26 committed:
updates

kartiks26 committed:
Merge pull request #425 from navgurukul/uiUpdateV2

updates

kartiks26 committed:
leaning track work

kartiks26 committed:
Merge branch 'dev' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

amansharmma committed:
Bug Fixed

kartiks26 committed:
done with learning Tracks

kartiks26 committed:
Merge pull request #433 from navgurukul/uiUpdateV2

done with learning Tracks

kartiks26 committed:
Merge branch amans Updates

kartiks26 committed:
Merge pull request #434 from navgurukul/uiUpdateV2

UI update v2

kartiks26 committed:
fixed Learning Track Bugs

kartiks26 committed:
Merge pull request #435 from navgurukul/uiUpdateV2

fixed Learning Track Bugs

amansharmma committed:
Fixed some Changes on the Card on the My learning track page

jschanker commented:
On small screens, text can get cut off and overlap logo (https://bhanwari-devi.vercel.app/Returning-user):

![image](https://user-images.githubusercontent.com/9159008/172405819-d45987c3-12b4-4c95-a8e7-8c6cd443024d.png)

jschanker reviewed (commented):
Some initial comments.

kartiks26 committed:
narendra feedbacks

kartiks26 committed:
Merge branch 'classBatch' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

kartiks26 committed:
Merge pull request #437 from navgurukul/AmanclassBatchUiFixes

Amanclass batch UI fixes

kartiks26 committed:
Merge branch 'classBatch' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

kartiks26 committed:
Merge pull request #438 from navgurukul/uiUpdateV2

UI update v2

jschanker reviewed (commented):
Some more comments

kartiks26 committed:
updated Alert Box For diffrent type of enrollments

amansharmma committed:
Fixed Card changes  on the My learning track page

amansharmma committed:
Fixed Card

kartiks26 committed:
updated the class click navigation

kartiks26 committed:
Merge pull request #441 from navgurukul/uiUpdateV2

UI update v2

kartiks26 committed:
Merge branch 'AmanclassBatchUiFixes' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

kartiks26 committed:
Merge pull request #442 from navgurukul/uiUpdateV2

UI update v2

amansharmma committed:
Worked to Video changes and Feedback from chankar sir

amansharmma committed:
Merge branch 'classBatch' of https://github.com/navgurukul/bhanwari-devi into AmanclassBatchUiFixes

kartiks26 committed:
Merge pull request #448 from navgurukul/AmanclassBatchUiFixes

Amanclass batch UI fixes

jschanker committed:
Replace dateTimeFormat implementation with date-fns's

And updated references to it in the classBatch branch
* Placed the function implementation in date-fns and also added doc comments to functions

kartiks26 committed:
Merge pull request #449 from jschanker/chore/migrate-to-date-fns-classBatch

Chore/Use date-fns for dateTimeFormat

amansharmma committed:
Added course completionPage

jschanker committed:
Create/fix ClassJoinTimerButton, make Date functionality more robust

* Added and improved Date functionality, which included converting IST timestamp to ISOString when passing to Date constructor for browser consistency (As per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#timestamp_string :
"When parsing date strings with the Date constructor (and Date.parse, they are equivalent), always make sure that the input conforms to the ISO 8601 format (YYYY-MM-DDTHH:mm:ss.sssZ) — the parsing behavior with other formats is implementation-defined and may not work across all browsers. Support for RFC 2822 format strings is by convention only.")
* Factored out common ClassJoinTimerButton from ClassCard/Volunteer form
* Fix: Put creation of timer in useEffect and cleaned up so a new one wasn't created with each passing minute and so all the repeating timers didn't remain after navigating to a different page.

kartiks26 committed:
updated some minor bugs

kartiks26 committed:
Merge branch 'CourseCompletion' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

kartiks26 commented:
@jschanker  is review for code is pending or is done .

kartiks26 committed:
Merge branch 'main' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

jschanker reviewed (commented):
@kartiks26 Wasn't done with the review yet.  Will this be merged in soon?  I had started looking at centralizing `Date` related functionality because I thought this was still in progress but I'll return to reviewing this PR more.  Here are some more review comments in the meantime.

kartiks26 committed:
fixed redux issue and doubt class issue

kartiks26 committed:
Merge pull request #462 from navgurukul/uiUpdateV2

UI update v2

kartiks26 commented:
@jschanker  Earlier, This PR didn't contain Pull from main and was not in cordination with the latest review fixes and updates to main so basically it was behind some updates .
now today i merged main into the classBatch branch in here so it might reduce some errors there were in the other component which was intially behind
There is plan to update all component into MUI so component which are not in MUI can be neglated as they will eventually be updated with new code.

jschanker reviewed (commented):
Few more comments

Poonam-Singh-Bagh committed:
assessment

kartiks26 committed:
Merge branch 'uiUpdateV2' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

kartiks26 committed:
code review

amansharmma committed:
Resolved PR Feedback & Fixed Button

kartiks26 committed:
Merge pull request #464 from navgurukul/ClassBatchBugs

Class batch bugs

kartiks26 committed:
fixed in exercise enroll issue

kartiks26 committed:
fixed alot of ui and other minor bugs

kartiks26 committed:
Merge pull request #465 from navgurukul/uiUpdateV2

UI update v2

kartiks26 committed:
adas

kartiks26 committed:
fixed some issues

kartiks26 committed:
Merge pull request #466 from navgurukul/uiUpdateV2

UI update v2

kartiks26 committed:
Merge pull request #468 from navgurukul/classBatch

Class batch

Poonam-Singh-Bagh committed:
Merge branch 'main' of https://github.com/navgurukul/bhanwari-devi into dev

jschanker committed:
Allow joining at 10 minutes before

* Added differenceInMinutes/minutesUntil to date.js
* Now use minutesUntil to more directly check if student can join instead of checking the resulting string from timeLeftFormat
* Clean up repeating timer when the student can join

kartiks26 committed:
done with learning track stuff

jschanker committed:
Fix file name, start unit tests, revert state to message

* Renamed ClassJoinTimerButton.js => index.js so imports work
* Reverted to setting timeRemainingMsg as state since canJoin will only update once whereby no re-renders to update timer will be done
* Called setTimeRemainingMsg in useEffect outside setInterval callback so the time will be updated immediately upon change of start time (as opposed to waiting a minute when setInterval would cause it to change)

jschanker committed:
Tests for disabled before/enabled after allowed join time

Poonam-Singh-Bagh committed:
Showed classes according to date and pathway

jschanker committed:
Refactor tests/use forEach instead of for loop

kartiks26 committed:
completed learning track

kartiks26 committed:
Merge pull request #473 from navgurukul/uiUpdateV2

Done with Learning track

kartiks26 committed:
Merge pull request #474 from navgurukul/classBatch

Class batch

Poonam-Singh-Bagh committed:
Merge branch 'main' of https://github.com/navgurukul/bhanwari-devi into bug/volunteer-automation

Poonam-Singh-Bagh committed:
changed volunteer flow route

Poonam-Singh-Bagh committed:
changed volunteer flow route

Poonam-Singh-Bagh committed:
volunteer automation

Poonam-Singh-Bagh committed:
volunteer automation

Poonam-Singh-Bagh committed:
Merge pull request #476 from navgurukul/bug/volunteer-automation-dev

Bug/volunteer automation dev

kartiks26 committed:
Merge pull request #477 from navgurukul/LearningTrackVideo

Learning track video

jschanker committed:
Merge branch 'classBatch' into class-join-timer-button-and-improved-date-functionality

jschanker committed:
Change moment to format for ClassCard

jschanker committed:
Merge pull request #460 from jschanker/class-join-timer-button-and-improved-date-functionality

New Class join timer button component and improved date functionality

jschanker committed:
Use date-fns format instead of moment format (IndividualVolunteers)

Resolves https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r894094593 and https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r894095272 by using user's local time.

jschanker committed:
Use date-fns format instead of moment format (VolunteerDetails)

jschanker committed:
Chore/Add GitHub action to test on every push/pull request

https://github.com/navgurukul/bhanwari-devi/pull/471

jschanker committed:
Avoid duplicate build tests/use npm install

* Previously, tests were being run twice for pull request/commit.
* npm install is necessary if package-lock.json is not up to date.

jschanker committed:
Commented out outdated replacement for timestamps

Times are currently received in IST, not UTC, and date processing is being moved to `src/common/date.js`.

amansharmma committed:
FIxed some Course completion Mobile view styling

amansharmma committed:
Added the video in the pathwaypage

kartiks26 committed:
Merge pull request #479 from navgurukul/LearningTrackVideo

Added the video in the pathwaypage

kartiks26 committed:
implemented jason TimerComponent

kartiks26 committed:
Merge pull request #480 from navgurukul/uiUpdateV2

UI update v2

kartiks26 committed:
Merge branch 'CourseCOmpletionUpdate' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

kartiks26 committed:
Merge pull request #481 from navgurukul/uiUpdateV2

UI update v2

jschanker commented:
Since suggestions are getting hidden, here are some ones that appear to be unresolved on the currently published classBatch branch at least:
https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r891112463, https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r891339459 (IMPORTANT if it's an issue), https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r895772979 (may be too much work to resolve now, but just to keep in mind), https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r895781369, https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r895799733, https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r895806727, https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r896266462, https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r897607717, https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r897819115

jschanker commented:
Also, the PATH comments: https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r897591638, https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r897592665, https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r897596847, https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r897597263, https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r897597678

jschanker committed:
Add isBefore/isBeforeNow, use isBeforeNow instead of TimeLeft

jschanker reviewed (commented):
More comments

amansharmma committed:
Added deferent video in each pathway

jschanker commented:
Having difficulty recreating now, but somehow there were no batches and errors ensued due to this bad request: 
> XHR failed loading: GET "https://dev-api.merakilearn.org//pathways/undefined/courses?courseType=json"

![image](https://user-images.githubusercontent.com/9159008/175458835-86bc168e-7fcd-43f2-a8e9-d831a0b5f653.png)

jschanker committed:
Fix/use wrapper isBefore to handle date strings

kartiks26 commented:
> Since suggestions are getting hidden, here are some ones that appear to be unresolved on the currently published classBatch branch at least: [#431 (comment)](https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r891112463), [#431 (comment)](https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r891339459) (IMPORTANT if it's an issue), [#431 (comment)](https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r895772979) (may be too much work to resolve now, but just to keep in mind), [#431 (comment)](https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r895781369), [#431 (comment)](https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r895799733), [#431 (comment)](https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r895806727), [#431 (comment)](https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r896266462), [#431 (comment)](https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r897607717), [#431 (comment)](https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r897819115)

@Poonam-Singh-Bagh

kartiks26 commented:
> Also, the PATH comments: [#431 (comment)](https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r897591638), [#431 (comment)](https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r897592665), [#431 (comment)](https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r897596847), [#431 (comment)](https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r897597263), [#431 (comment)](https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r897597678)

@Poonam-Singh-Bagh

kartiks26 committed:
added the dashboard in header

kartiks26 committed:
Merge pull request #486 from navgurukul/uiUpdateV2

added the dashboard in header

jschanker committed:
New BeforeOrAfter component, use for ClassBatch, Tests

jschanker committed:
Fix/Remove previous && from ClassTopic

jschanker committed:
Fix/before given date means after now so use !isBeforeNow

* Also use differenceInMilliseconds in millisecondsUntil, not seconds in date.js
* Initialize isBefore outside setTimeout in useEffect so no delay in scheduling state update when the date is changed
* Update tests to correct for errors, mainly that a time before now should show the after element, after now the before element

jschanker committed:
Change to BeforeOrAfter to FutureOrPast, fix import

jschanker committed:
Reduce distance to dates to avoid TimeoutOverflowWarning

kartiks26 committed:
Merge pull request #488 from jschanker/feat/BeforeOrAfter-for-classBatch

Feat/FutureOrPast component for class batch

kartiks26 committed:
resolved comments

kartiks26 committed:
Merge branch 'classBatch' of github.com:navgurukul/bhanwari-devi into uiUpdateV2

kartiks26 committed:
resolved comments

kartiks26 committed:
Merge pull request #490 from navgurukul/uiUpdateV2

UI update v2

kartiks26 committed:
Merge pull request #485 from navgurukul/LearningTrackVideo

Added deferent video in each pathway

Poonam-Singh-Bagh committed:
class & batch

Apekshahande committed:
added css

jschanker reviewed (commented):
Some minor styling questions about paths
